public/scripts/mapScript_gameOverview.js

```diff
@@ -77,16 +83,131 @@ function initMap() {
     const tiles = L.tileLayer(tileURL, { attribution });
 
     tiles.addTo(mymap);
+    mymap.on('click', addLoot);
+}
+
+function setGameId(game_id) {
+    gameId = game_id;
 }
 
-function applyLootMarker(lat, lng, loot_name) {
+function applyLootMarker(lat, lng, loot_name, loot_id) {
     let latlng = L.latLng(lat, lng);
     let newMarker = L.marker(latlng, {icon: lootIcon})
         .bindPopup(L.popup({ maxWidth: maxPopupWidth })
             .setContent('Buit: ' + loot_name))
         .addTo(mymap);
+    newMarker.on('click', function (e) {
+        selectedLootId = loot_id;
+        remove_loot_button.textContent = 'Verwijder buit: ' + loot_name;
+        remove_loot_button.disabled = false;
+    });
+    applyEvents(newMarker);
+    lootMarkers.push(newMarker);
+    lootLatLngs.push(newMarker.getLatLng());
+    lootIds.push(parseInt(loot_id));
+}
+
+function addLoot(e) {
+    if (!lootNameInput.value || lootNameInput.value.trim() === '') {
+        if (mapBox.children.length > 3) {
+            return;
+        }
+        let errorMsg = document.createElement('p');
+        errorMsg.style.color = 'red';
+        errorMsg.textContent = 'Vul a.u.b. een naam in voor de buit.';
+        mapBox.appendChild(errorMsg);
+
+        setTimeout(() => {
+            if (mapBox.children.length > 3) {
+                mapBox.removeChild(errorMsg);
+            }
+        }, 5000);
+        return;
+    }
+
+    let contains = false;
+    lootLatLngs.forEach(latlng => {
+        if (latlng.equals(e.latlng)) {
+            contains = true;
+        }
+    });
+    if (contains) {
+        return;
+    }
+    let newMarker = L.marker(e.latlng, { icon: lootIcon })
+        .bindPopup(L.popup({ maxWidth: maxPopupWidth})
+            .setContent('Buit: ' + lootNameInput.value.trim()))
+        .addTo(mymap);
     applyEvents(newMarker);
     lootMarkers.push(newMarker);
+    lootLatLngs.push(newMarker.getLatLng());
+    saveLoot(gameId, lootNameInput.value.trim());
+}
+
+async function saveLoot(game_id, loot_name) {
+    if (lootMarkers.length < 1) {
+        return;
+    }
+    $.ajaxSetup({
+        headers: {
+            'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+        }
+    });
```

Comment:
Dit zou ik plaatsen in een standaard functie, want ik zag later in het script dat het nogmaals gebeurt.

Reply:
Dat kan, ik zie het nut er alleen niet stevig van in